GitUI/Editor/RichTextBoxXhtmlSupportExtension.cs

```diff
@@ -1146,6 +1101,85 @@ public static string GetPlainText(this RichTextBox rtb, int from, int to)
             return text.ToString();
         }
 
+        public static string GetLink(this RichTextBox rtb, int charIndex)
+        {
+            var text = rtb.Text;
+            if (charIndex < 0 || text.Length <= charIndex)
+            {
+                return null;
+            }
+
+            IntPtr oldMask = rtb.BeginUpdate();
+
+            int nStart = rtb.SelectionStart;
+            int nEnd = rtb.SelectionLength;
+
+            try
+            {
+                //--------------------------------
+                // this is an inefficient method to get text format
+                // but RichTextBox doesn't provide another method to
+                // get something like an array of charformat and paraformat
+                //--------------------------------
+                rtb.Select(charIndex, 1);
+                if (!rtb.IsLink())
+                {
+                    return null;
+                }
+
+                int from = charIndex;
+                while (from > 0)
+                {
+                    rtb.Select(from - 1, 1);
+                    if (!rtb.IsLink())
+                    {
+                        break;
+                    }
+
+                    --from;
+                }
+
+                int to = charIndex + 1;
+                while (to < text.Length)
+                {
+                    rtb.Select(to, 1);
+                    if (!rtb.IsLink())
+                    {
+                        break;
+                    }
+
+                    ++to;
+                }
+
+                if (to < text.Length && text[to] == '#')
+                {
+                    ++to;
+                    from = to;
+                    while (to < text.Length && !char.IsWhiteSpace(text[to]) && text[to] != ',')
+                    {
+                        ++to;
+                    }
+                }
+
+                return text.Substring(from, to - from);
+            }
+            catch (Exception /*ex*/)
+            {
+                ////MessageBox.Show(ex.Message);
```

Comment:
I would return `null` here instead of line 1180, as it feels more logical.
Also please remove the commented code.

Reply:
Yes, copied from an other function and not tought to the end...